frontend/src/libs/packages/store/middlewares/middlewares.ts

```diff
@@ -1,16 +1,22 @@
 import { type AnyAction, type Middleware } from '@reduxjs/toolkit';
 import { isRejected } from '@reduxjs/toolkit';
+import { ExceptionMessage } from 'shared/build/index.js';
 
 import { NotificationType } from '#libs/packages/notification/libs/enums/enums.js';
 import { type AppDispatch } from '#libs/packages/store/libs/types/types.js';
 import { appActions } from '#slices/app/app-notification.js';
+import { signOut } from '#slices/auth/actions.js';
 
 const handleError: Middleware<unknown, unknown, AppDispatch> = ({
   dispatch,
 }) => {
   return (next) => {
     return (action: AnyAction) => {
       if (isRejected(action)) {
+        if (action.error.message === ExceptionMessage.UNAUTHORIZED_USER) {
```

Comment:
We need to listen on 401 in BaseHttpApi and throw custom error and catch it here

Reply:
Also please create handleUnauthorized for decompose error handling

Reply:
@v0ldemar01 clarify for me please, so I understand it correctly.: 
1. I need to throw an HttpError in the checkResponse or handleError of BaseHttpApi? 
2. Then I need to create handle-unauthorized.middleware.ts for example? 
3. If yes, do I need to create separate file for handleError middleware and make re-exports inside of the middleware.ts? And then add them inside of the store package?

Reply:
It's not a HttpError, a custom one that extends from Error, as for handle-unauthorized.middleware.ts you are right, just Implement as you see fit